Patch-Source: https://github.com/calf-studio-gear/calf/commit/f6c6aaef3d8c744faefa13842fd452323bb30750
--
From cec78130b8ea1fcd7fe8c2f93de004df134e6ce5 Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Thu, 21 Sep 2023 19:08:39 -0400
Subject: [PATCH] Fix build with LLD

LLVM's LLD handles the -retain-symbols-file option (used by
-export-symbols-regex in libtool) differently from GNU ld, causing
undefined references during link. This commit removes the
-export-symbols-regex option from libcalf_la_LDFLAGS since by default
libtool exports all symbols anyway, so it should not be necessary.

Fixes https://github.com/calf-studio-gear/calf/issues/156.

Signed-off-by: Violet Purcell <vimproved@inventati.org>
---
 src/Makefile.am | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index b1e37c522..6f11628ba 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -39,11 +39,7 @@ calfbenchmark_LDADD = libcalf.la
 
 calf_la_SOURCES = audio_fx.cpp analyzer.cpp lv2wrap.cpp metadata.cpp modules_tools.cpp modules_delay.cpp modules_comp.cpp modules_limit.cpp modules_dist.cpp modules_filter.cpp modules_mod.cpp modules_pitch.cpp fluidsynth.cpp giface.cpp monosynth.cpp organ.cpp osctl.cpp plugin.cpp preset.cpp synth.cpp utils.cpp wavetable.cpp modmatrix.cpp
 calf_la_LIBADD = $(FLUIDSYNTH_DEPS_LIBS) $(GLIB_DEPS_LIBS) 
-if USE_DEBUG
 calf_la_LDFLAGS = -rpath $(pkglibdir) -avoid-version -module -lexpat -disable-static
-else
-calf_la_LDFLAGS = -rpath $(pkglibdir) -avoid-version -module -lexpat -disable-static -export-symbols-regex "lv2_descriptor"
-endif
 
 if USE_LV2_GUI
 
